ffi/diplomat/src/timezone.rs

```diff
@@ -33,143 +22,123 @@ pub mod ffi {
     use icu_timezone::TimeZoneBcp47Id;
     use icu_timezone::ZoneVariant;
 
-    #[diplomat::rust_link(icu::timezone::GmtOffset, Struct)]
-    pub struct ICU4XGmtOffset {
-        pub(super) _private_data: i32,
-    }
-
-    pub struct ICU4XOptionalGmtOffset {
-        pub(super) _private_data: i32,
-    }
-
+    #[diplomat::opaque]
     #[diplomat::rust_link(icu::timezone::CustomTimeZone, Struct)]
-    pub struct ICU4XCustomTimeZone {
-        pub gmt_offset: ICU4XOptionalGmtOffset,
-        pub time_zone_id: ICU4XOptionalTinyAsciiStr8,
-        pub metazone_id: ICU4XOptionalTinyAsciiStr4,
-        pub zone_variant: ICU4XOptionalTinyAsciiStr2,
-    }
+    pub struct ICU4XCustomTimeZone(pub CustomTimeZone);
 
     #[diplomat::opaque]
     #[diplomat::rust_link(icu::timezone::MetaZoneCalculator, Struct)]
     pub struct ICU4XMetaZoneCalculator(pub MetaZoneCalculator);
 
-    impl ICU4XGmtOffset {
-        pub fn create_from_str(s: &str) -> DiplomatResult<ICU4XGmtOffset, ICU4XError> {
-            GmtOffset::from_str(s)
-                .map(Into::into)
+    impl ICU4XCustomTimeZone {
+        pub fn create_from_str(s: &str) -> DiplomatResult<Box<ICU4XCustomTimeZone>, ICU4XError> {
+            CustomTimeZone::from_str(s)
+                .map(ICU4XCustomTimeZone::from)
+                .map(Box::from)
                 .map_err(Into::into)
                 .into()
         }
 
         #[diplomat::rust_link(icu::timezone::GmtOffset::try_from_offset_seconds, FnInStruct)]
-        pub fn try_from_offset_seconds(seconds: i32) -> DiplomatResult<ICU4XGmtOffset, ICU4XError> {
-            GmtOffset::try_from_offset_seconds(seconds)
-                .map(Into::into)
-                .map_err(Into::into)
-                .into()
+        pub fn try_set_gmt_offset_seconds(
+            &mut self,
+            offset_seconds: i32,
+        ) -> DiplomatResult<(), ICU4XError> {
+            match GmtOffset::try_from_offset_seconds(offset_seconds) {
+                Ok(v) => {
+                    self.0.gmt_offset = Some(v);
+                    Ok(())
+                }
+                Err(e) => Err(ICU4XError::from(e)),
+            }
+            .into()
         }
 
-        #[diplomat::rust_link(icu::timezone::GmtOffset::offset_seconds, FnInStruct)]
-        pub fn offset_seconds(&self) -> i32 {
-            GmtOffset::from(self).offset_seconds()
+        pub fn gmt_offset_seconds(&self) -> DiplomatResult<i32, ()> {
+            self.0
+                .gmt_offset
+                .map(|v| v.offset_seconds())
+                .ok_or(())
+                .into()
         }
 
         #[diplomat::rust_link(icu::timezone::GmtOffset::is_positive, FnInStruct)]
-        pub fn is_positive(&self) -> bool {
-            GmtOffset::from(self).is_positive()
+        pub fn is_gmt_offset_positive(&self) -> DiplomatResult<bool, ()> {
```

Comment:
nit: docs mentioning what the result is doing

Reply:
5eb2d73